pkg/manager/modelregistry.go

```diff
@@ -79,6 +91,82 @@ func (m *Manager) Capabilities() []*gnmi.ModelData {
 	return outputList
 }
 
-func toModelName(name string, version string) string {
+// extractReadOnlyPaths is a recursive function to extract a list of read only paths from a YGOT schema
+func extractReadOnlyPaths(deviceEntry *yang.Entry, parentState yang.TriState, parentPrefix string, parentPath string) []string {
+	readOnlyPaths := make([]string, 0)
+
+	for _, dirEntry := range deviceEntry.Dir {
+		prefix := ""
+		if dirEntry.Prefix != nil {
+			prefix = dirEntry.Prefix.Name
+		}
+		var itemPath string
+		if parentPath == "" {
+
+			itemPath = fmt.Sprintf("%s/%s:%s", parentPath, prefix, dirEntry.Name)
```

Comment:
should format name be use also at 106 and 108 ? also 105 is an empty useless line

Reply:
don't know what I was thinking of - all consolidated now